content/Link Tracking with StimulusReflex - Part II.md

```diff
@@ -0,0 +1,613 @@
+# Link Tracking with StimulusReflex - Part II
+
+In the [first part of this series](TODO: Add link to first part) we took care of setting up our project to use Stimulus Reflex. Now it's time to get our hands dirty and write some code. The first thing we are going to do is take care of managing our shortened links.
+
+![Header image](images/stimulus_reflex.jpeg)
+
+## Creating the model
+
+Let's start with some good old fashioned Rails code, we'll create the migration for our links and the associated model:
+
+
+```ruby
+class CreateShortenedLinks < ActiveRecord::Migration[6.0]
+  def change
+    create_table :shortened_links do |t|
+      t.string :name, null: false
+      t.string :shortened_path, null: false
+      t.string :original_url, null: false
+      t.integer :views_count, default: 0
+
+      t.timestamps
+    end
+  end
+end
+```
+
+```ruby
+# app/models/shortened_link
+
+class ShortenedLink < ApplicationRecord
+  validates :name, :shortened_path, :original_url, presence: true
+  validates :shortened_path, uniqueness: true
+end
+```
+
+Notice how we added some simple validations to make it more fun.
+
+## Listing links
+
+Let's start by the simplest of actions: listing the links. To do so, we'll create the following files:
+
+```ruby
+# app/controllers/shortened_links_controller.rb
+
+class ShortenedLinksController < ApplicationController
+  def index
+    @shortened_links ||= ShortenedLink.all
```

Comment:
why the `||` ?

Reply:
we don't want to load new shortened links created by some other session?

Reply:
It's because StimulusReflex will re-run the controller action after every reflex. We want variables set on the reflex to take precedence. I think there's a way to avoid running controller actions by choosing the right morph but this is already pretty lengthy. I'm thinking of doing it on a Refactor article later on